modules/db/src/main/scala/gem/dao/ObservationDao.scala

```diff
@@ -35,98 +35,161 @@ object ObservationDao {
            }.void
     } yield ()
 
-  /** Construct a program to select the specified observation, with the
-    * instrument but not targets nor steps.
-    */
-  def selectFlat(id: Observation.Id): ConnectionIO[(String, Instrument)] =
+  /**
+   * Construct a program to select the specified observation, with the
+   * instrument but not targets nor steps.  Raises an exception if the indicated
+   * observation does not exist.
+   */
+  def fetchFlat(id: Observation.Id): ConnectionIO[(String, Instrument)] =
     Statements.selectFlat(id).unique
 
-  /** Construct a program to select the specified observation, with the
-    * targets and instrument type but not steps.
-    */
-  def selectTargets(id: Observation.Id): ConnectionIO[(String, TargetEnvironment)] =
+  /**
+   * Construct a program to query for the specified observation, with the
+   * instrument but not targets nor steps.
+   */
+  def queryFlat(id: Observation.Id): ConnectionIO[Option[(String, Instrument)]] =
+    Statements.selectFlat(id).option
+
+  /**
+   * Construct a program to select the specified observation, with the
+   * targets and instrument type but not steps.  Raises an exception if the
+   * indicated observation does not exist.
+   */
+  def fetchTargets(id: Observation.Id): ConnectionIO[(String, TargetEnvironment)] =
     for {
-      o <- selectFlat(id)
+      o <- fetchFlat(id)
       t <- TargetEnvironmentDao.selectObs(id)
     } yield (o._1, t)
 
-  /** Construct a program to select the specified observation, with static
-    * config but not targets nor steps.
-    */
-  def selectStatic(id: Observation.Id): ConnectionIO[(String, StaticConfig)] =
+  /**
+   * Construct a program to query for the specified observation, with the
+   * targets and instrument type but not steps.
+   */
+  def queryTargets(id: Observation.Id): ConnectionIO[Option[(String, TargetEnvironment)]] =
+    for {
+      o <- queryFlat(id)
+      t <- o.as(TargetEnvironmentDao.selectObs(id)).sequence
+    } yield o.map(_._1).product(t)
```

Comment:
I thought this was `fproduct`

Reply:
`fproduct` from `Functor` takes an `F[A]` and a function `A => B` and creates a tuple `F[(A, B)]` while `product` from `Apply` just tuples two values in `F` together: 

```
def product[A, B](fa: F[A], fb: F[B]): F[(A, B)]
```

here `F` is `Option` of course.